QuoridorEngine/src/core/QuoridorGame.cs

```diff
@@ -196,6 +196,109 @@ private void placeWall(QuoridorMove move)
             gameHistory.Add(move);
         }
 
+        /// <summary>
+        /// Check whether a path connecting current player's square
+        /// and his target baseline exists using DFS algorithm.
+        /// </summary>
+        /// <param name="isWhite">Current player is White</param>
+        /// <returns>True if path exists</returns>
+        private bool playerCanReachBaseline(bool isWhite)
+        {
+            // Pending squares to be explored
+            Stack<(int, int)> frontierSquares = new();
+
+            // Already visited squares
+            HashSet<(int, int)> visitedSquares = new();
+
+            // Get current player object
+            QuoridorPlayer currentPlayer = getTargetPlayer(isWhite);
+            Debug.Assert(currentPlayer != null);
+
+            // Temporarily save player's current row
+            int initialRow = currentPlayer.Row;
+
+            // Add player's current square in frontier as first square to be explored
+            frontierSquares.Push((currentPlayer.Row, currentPlayer.Column));
+            while (frontierSquares.Count != 0)
+            {
+                // Get a square (and remove it) from frontier
+                (int currentSquareRow, int currentSquareCol) = frontierSquares.Pop();
+
+                // Skip this square if it has already been visited
+                if (visitedSquares.Contains((currentSquareRow, currentSquareCol))) 
+                    continue;
+
+                // Store current square as visited
+                visitedSquares.Add((currentSquareRow, currentSquareCol));
+
+                // Temporarily update player's current row 
+                currentPlayer.Row = currentSquareRow;
```

Comment:
Maybe create a getter for the target baseline in Player instead of temporarily moving the player. Alternatively, if you do not want to break the encapsulation add a new method in QuoridorPlayer called RowIsTheTargetBaseline(int row) and use that instead.

In my opinion the second option is the better one (RowIsTheTargetBaseline)